pkg/kubelet/kubelet.go

```diff
@@ -875,7 +883,25 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		kubeDeps.Recorder,
 		volumepathhandler.NewBlockVolumePathHandler())
 
-	klet.backOff = flowcontrol.NewBackOff(backOffPeriod, MaxContainerBackOff)
+	max := MaxContainerBackOff
+	base := containerBackOffPeriod
+	if utilfeature.DefaultFeatureGate.Enabled(features.KubeletCrashLoopBackOffMax) {
+		if kubeCfg.CrashLoopBackOff.MaximumBackOffPeriod != nil {
```

Comment:
Here we need to check if `kubeCfg.CrashLoopBackOff != nil`, If you modify it to be non-pointer or add a default value according to thockin's suggestion, you can avoid this.